libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/OpenGroupPoller.kt

```diff
@@ -86,22 +170,30 @@ class OpenGroupPoller(private val server: String, private val executorService: S
                     isCaughtUp = true
                 }
             }
-            executorService?.schedule(this@OpenGroupPoller::poll, pollInterval, TimeUnit.MILLISECONDS)
+
+            // Only poll again if it's the same poller run
+            if (currentRunId == runId) {
+                future = executorService?.schedule(this@OpenGroupPoller::poll, pollInterval, TimeUnit.MILLISECONDS)
+            }
         }.fail {
-            updateCapabilitiesIfNeeded(isPostCapabilitiesRetry, it)
+            updateCapabilitiesIfNeeded(isPostCapabilitiesRetry, currentRunId, it)
         }.map { }
```

Comment:
useless `map{}`, right?

Reply:
The map was there to convert the return value from `Promise<List<OpenGroupApi.BatchResponse<*>>, Exception>` to `Promise<Unit, Exception>`